### PR TITLE
Remove non-serializable objects from error.userData when request fails

### PR DESCRIPTION
**Pull request by akshetpandey**
error.userData may contain NSURL which doesn't serialize and causes a crash.

### CODE REVIEW

File: InAppUtils/InAppUtils.m [hunk 1]
```diff
@@ -220,12 +220,18 @@ - (void)productsRequest:(SKProductsRequest *)request
...
-- (void)request:(SKRequest *)request didFailWithError:(NSError *)error{
+- (void)request:(SKRequest *)request didFailWithError:(NSError *)error
+{
     NSString *key = RCTKeyForInstance(request);
     RCTResponseSenderBlock callback = _callbacks[key];
-    if(callback) {
+    if (callback) {
+        // Ensure error.userData can be converted to JSON without error.
+        // This will remove any NSURL from the error.userData.
+        error = RCTErrorClean(error);
```
Comment (chirag04) [inline InAppUtils/InAppUtils.m]: this should be upstream in react-native? they should handle NSURL in `RCTJSErrorFromNSError`?

Comment (akshetpandey) [inline InAppUtils/InAppUtils.m]: Ideally, but upstreaming it and waiting for merge will take pretty long and till then this method will crash.

I will try upstream that change too

Comment (chirag04) [inline InAppUtils/InAppUtils.m]: agreed. let's get a PR upstream. once it's merged upstream, we can add a fallback here. If it's not merged upstream in x time, then let's ship here.

that works for you?

**Comment by chirag04** [comment]
@akshetpandey any update?

**Comment by chirag04** [comment]
@akshetpandey ping

**Pull request closed without merging**